draft-chen-cfrg-vdaf-pine.md

```diff
@@ -478,17 +478,77 @@ and `PineValid.eval_wr_check()`, and evaluate it at the final joint randomness
-> algorithms will come later, in a future draft. If you're using too many
-> variable names, then you're probably not on the right track.
+This section describes PINE VDAF for {{ROCT23}}, a one-round VDAF with no
+aggregation parameter. It takes a set of Client gradients expressed as vectors
+of floating point values, and computes an element-wise summation of valid
+gradients with bounded L2-norm configured by the user parameters in
+{{pine-user-param}}. The VDAF largely uses the encoding and validation schemes
+in {{flp}}, and also specifies how the joint randomness shared between the
+Client and Aggregators is derived. There are two kinds of joint randomness used:
+
+* "Verification joint randomness": These are the field elements used by the
+  Client and Aggregators to evaluate the FLP circuit. The verification joint
+  randomness is derived similar to the joint randomness in Prio3
+  {{Section 7.2.1.2 of !VDAF}}: the XOF is applied to each secret share of the
+  encoded measurement to derive the "part"; and the parts are hashed together,
+  using the XOF once more, to get the seed for deriving the joint randomness
+  itself.
+* "Wraparound joint randomness": This is used to generate the random vectors in
+  the wraparound checks that both the Clients and Aggregators need to derive on
+  their own. It is generated in much the same way as verification joint
+  randomness, except that only the gradient and the range-checked norm are used
+  to derive the parts.
+
+In order for the Client to shard its gradient into input shares for the
+Aggregators, the Client first encodes its gradient into field elements, and
+encodes the range-checked L2-norm, according to {{encode-gradient-and-norm}}.
+Then it needs to derives the wraparound joint randomness for the wraparound
+checks. It first derives the parts, by applying the XOF to the secret shares of
+the encoded gradient and range-checked norm. Then it applies the XOF again to
+the parts to obtain the seed for the wraparound joint randomness and use it to
+encode the results for wraparound checks {{encode-wr-check}}. The encoded
+gradient, range-checked norm, and results for wraparound checks form the fully
+encoded measurement, that will be secret-shared to (1) be sent as input shares
+for the Aggregators, and (2) derive the verification joint randomness parts.
+The parts are then hashed by the XOF to generate the seed for the verification
+joint randomness to evaluate the FLP circuit and generate the proof. The proof
+is also secret-shared, and sent to the Aggregators, along with the input shares,
+and the joint randomness parts for both wraparound and verification joint
+randomness.
+
+Then the Aggregators carry out a multi-party computation to obtain the output
+shares (the secret shares of the encoded Client gradients), and also reject
+Client gradients that have invalid L2-norm. Each Aggregator first needs to
+derive wraparound and verification joint randomness. To compute the XOF seed
```

Comment:
Like the last paragraph, avoid saying things twice. Make sure to highlight the differences between what the Client computes versus what each Aggregator computes.

Reply:
Done.